crates/consensus/src/transaction/legacy.rs

```diff
@@ -91,11 +91,11 @@ impl TxLegacy {
         signature.write_rlp_vrs(out);
     }
 
-    /// Output the length of the RLP signed transaction encoding.
-    pub fn payload_len_with_signature(&self, signature: &Signature) -> usize {
+    /// Returns what the encoded length should be, if the transaction were RLP encoded with the
+    /// given signature.
+    pub(crate) fn encoded_len_with_signature(&self, signature: &Signature) -> usize {
```

Comment:
side note: i consider these all tech debt that is paid down by a refactor of rlp derives

Reply:
yes, rlp(flatten) etc will usher in a new era of derived and correct encodings